### PR TITLE
Map single values to List properties

### LINKED ISSUE #6448: ConfigMapper - multi value parameters with single value
Thanks to #6386 we are now able to parse a multi value paramter with multiple values from a thing file. However if we provide just a single value for such a parameter (multi value optional), the ConfigMapper tries to map a String value onto an ArrayList/Collection property of the config class.

Proposal: The objectConvert function of the ConfigMapper should provide a special case for a String value and a collection type.

Best regards
Daniel

### PR DESCRIPTION
**Pull request by htreu**
Fixes #6448.

Also: Adds ConfigMapper test.

Signed-off-by: Henning Treu <henning.treu@telekom.de>

### CODE REVIEW

File: bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/internal/ConfigMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     */
+    static class SimpleTestConfiguration {
+        String string;
+        Integer integer;
+        Boolean bool;
+        List<String> list;
+    }
+
+}
```
Comment (davidgraeff) [inline bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/internal/ConfigMapperTest.java]: There is already a test suite for ConfigMapper. See https://github.com/eclipse/smarthome/blob/ef748dc460f556e7dd373235282da6ab472191ba/bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/ConfigurationTest.java

Comment (htreu) [reply bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/internal/ConfigMapperTest.java]: Thanks, I will move my test case there.

File: bundles/config/org.eclipse.smarthome.config.core.test/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -14,6 +14,7 @@ Import-Package:
  org.eclipse.smarthome.test,
  org.eclipse.smarthome.test.java,
  org.hamcrest;core=split,
+ org.hamcrest.core,
```
Comment (davidgraeff) [inline bundles/config/org.eclipse.smarthome.config.core.test/META-INF/MANIFEST.MF]: Is that normal to have "org.hamcrest;core=split," and "org.hamcrest.core,"?

Comment (htreu) [reply bundles/config/org.eclipse.smarthome.config.core.test/META-INF/MANIFEST.MF]: Yes, two different statements.
@kaikreuzer ready to get merged.

**Pull request merged**